### PR TITLE
Variable Pickaxe Speeds

### PR DESCRIPTION
**Pull request by PerfectTangent**
### Purpose
This PR addresses the pickaxe half of #4901. Rather than mining speed being static, each tile can have a configurable mining speed and the quality of the pickaxe being used modifies that speed.

### Notes:

- Tiles have a configurable mining time. Default is the same as what the static mining time was (5 seconds).
- Pickaxes have a mining speed modifier
- Different pickaxes also had their damage numbers modified to mirror their /tg/ counterparts (most were already correct)
- This update also works with PickaxeMineable objects
- The Mining Vendor prefab now includes a single silver plated pickaxe for one lucky miner.

Tested smashing rocks and rock walls in editor, and it works. Thanks for the help, @corp-0!

### Changelog:

CL: Different tiers of pickaxes have varying mining speeds.

### CODE REVIEW

**Comment by PerfectTangent** [comment]
while I was looking through /tg/'s repo I found this gem in the mining_tools.dm

`//metal needed for a crowbar and for a knife, why the FUCK does a knife cost 6 metal sheets while a crowbar costs 0.025 sheets? shit makes no sense fuck this`

File: UnityProject/Assets/Scripts/Items/Tool/Pickaxe.cs [hunk 1]
```diff
@@ -8,6 +8,11 @@
 public class Pickaxe : MonoBehaviour, ICheckedInteractable<PositionalHandApply>
 {
 	[SerializeField] private AddressableAudioSource pickaxeSound = null;
+
+	[Tooltip("How much does this tool multiply mining time")] [SerializeField]
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Items/Tool/Pickaxe.cs:13]: Could you put this in the toolutil , so you just add the tool component to the pickaxe and make it call the toolutil function you told with the function you want to activate once it's finished
void ProgressComplete()
		{stuff you want to do when finished}
ServerUseToolWithActionMessages(interaction, default time, text for action, ect. ect. , ProgressComplete )

ServerUseToolWithActionMessages is a good example of the lambda function thing

Comment (PerfectTangent) [reply UnityProject/Assets/Scripts/Items/Tool/Pickaxe.cs:13]: Done, but there's some kind of issue with it that I can't identify.

Comment (NoooneyDude) [reply UnityProject/Assets/Scripts/Items/Tool/Pickaxe.cs:13]: Some kind of issue? Something preventing merge?

File: UnityProject/Assets/Scripts/Items/Tool/Pickaxe.cs [hunk 2]
```diff
@@ -25,40 +29,67 @@ public bool WillInteract(PositionalHandApply interaction, NetworkSide side)
...
-		var bar = StandardProgressAction.Create(ProgressConfig, ProgressComplete)
-			.ServerStartProgress(interaction.WorldPositionTarget.RoundToInt(),
-				5f, interaction.Performer);
-		if (bar != null)
-		{
-			SoundManager.PlayNetworkedAtPos(pickaxeSound, interaction.WorldPositionTarget, sourceObj: interaction.Performer);
-		}
+		ToolUtils.ServerUseToolWithActionMessages(
+			interaction, calculatedMineTime,
+			$"You start mining the {objectName}...",
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Items/Tool/Pickaxe.cs:56]: `objectName` isn't set. It _does_ get set in the other ServerPerformInteraction, however. You should be able to get the name of the tile from the interaction.

Comment (NoooneyDude) [reply UnityProject/Assets/Scripts/Items/Tool/Pickaxe.cs:56]: While you're at it, consider rescoping the `objectName` variable to within those methods; don't think it needs to be a class field.

Comment (corp-0) [reply UnityProject/Assets/Scripts/Items/Tool/Pickaxe.cs:56]: >  however. You should be able to get the name of the tile from the interaction

TargetObject is null for some reason which is preventing the tile region to trigger. Also you can't get the tile name from the interaction afaik, there is a ``DisplayName`` field in ``BasicTile`` but I rarely see it filled.

Comment (NoooneyDude) [reply UnityProject/Assets/Scripts/Items/Tool/Pickaxe.cs:56]: Right, tiles aren't objects. This is `PositionalHandApply`, ~you're probably after `TileApply`~. EDIT: oh, sorry, nevermind.

**Pull request merged**